### DISCUSSION TITLE
Mix MVC with Ajax

### DISCUSSION THREAD

**Discussion by hhvdblom** [dotnet/aspnetcore#22096] [General]
I don't get the technologies that Microsoft creates nowadays. Razor pages? Whats wrong with MVC? Blazor? Whats wrong with MVC. If you want a MVC application that behaves like a SPA you can change all anchor tags and form tags in Ajax calls. I wonder why nobody at Microsoft came up with that Idea. No load waiting like Angular and Blazor webassembly. All pages are loaded on demand. I created a main javascript module. it loads a couple of classes like History, Ajax. The Ajax class takes care for adding Ajax handlers  on anchor tags and form tags. 

The whole bullshit thing is that people are thinking the browser is in Swahili and the server is on the moon.  The web is nowadays fast enough to handle views with data.

**Comment by poke** [comment]
> Razor pages? Whats wrong with MVC?

Razor pages are actually _very similar_ to MVC. It’s just a different approach to the same thing, with a different focus on what the entry point to a page is. If you give it a try, you will see that it feels very natural for applications that focus a lot on content. But you will also see that it is still very similar to plain old MVC.

> Blazor? Whats wrong with MVC.

Nothing. But both serve a different purpose. Blazor is the answer to client-side web frameworks, which largely account for single-page applications. You can argue all day about that but while I certainly agree with that not everything _needs_ to be an SPA (and not everything _should_ be an SPA), there are cases that highly benefit from higher interactivity on the client side without requiring full roundtrips to the server.

But at the end, it’s all up to you and me, us application developers to select the best technology to fit the situation at hand. Sometimes, that will be plain MVC or Razor pages, sometimes we might need more stuff on the client with a SPA framework, and other times simple AJAX is the way to go. Nothing is off limit and all tools have their own merit. So don’t hate on other people’s choices and use what you think is best for your situation at hand.

**Reply by hhvdblom** [reply]
All I want to say is that I wanted to make a statement that Microsoft is creating lots of new technologies but does not make old technologies any better. MVC _could_  also easily be SPA. I created a ajax.js class with history in it so the application behaves as a SPA. I bet i will be much more productive as an so called Angular, or other crap framework developer. Microsoft should go that line because your are paying for visual studio. If you develop with Angular you don't need Visual Studio at all. The binding in MVC is already done for you, no need for Angular.

Microsoft nowadays is slow, very slow. If you look at where development is going to developers are realising they don't need Angular, Bootstrap and other crapware. They see the light and can develop big websites with Css Grid and Vanilla Javascript.
